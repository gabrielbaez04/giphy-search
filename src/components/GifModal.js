import React from 'react';
import Modal from 'react-modal';
import Slider from './Slider';
import PropTypes from 'prop-types';

class GifModal extends React.Component {
    componentWillMount() {
        Modal.setAppElement('body');
    }
    render() {
        const { gifs, selectedGif, modalIsOpen, onRequestClose } = this.props;
        return (
            selectedGif >= 0 &&
            <Modal className="gif-modal"
                isOpen={modalIsOpen}
                onRequestClose={() => onRequestClose()}>
                <Slider gifs={gifs} selectedGif={selectedGif} />
            </Modal>
        );
    }

};

GifModal.propTypes = {
    gifs: PropTypes.array.isRequired,
    selectedGif: PropTypes.number,
    modalIsOpen: PropTypes.bool.isRequired,
    onRequestClose: PropTypes.func.isRequired
}


export default GifModal;