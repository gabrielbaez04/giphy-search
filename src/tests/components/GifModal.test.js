import { shallow } from 'enzyme';
import GifModal from '../../components/GifModal';
import React from 'react';

let gifs, selectedGif, modalIsOpen, onRequestClose;

beforeEach(() => {
    gifs = [
        { id: 1, title: "test", images: { downsized_still: { url: "test.url.com" } } },
        { id: 2, title: "test", images: { downsized_still: { url: "test.url.com" } } },
        { id: 3, title: "test", images: { downsized_still: { url: "test.url.com" } } },
    ];
    selectedGif = 0;
    modalIsOpen = false;
    onRequestClose = jest.fn();
});
test('should render GifModal correctly', () => {
    const wrapper = shallow(<GifModal gifs={gifs} selectedGif={selectedGif} modalIsOpen={modalIsOpen} onRequestClose={onRequestClose} />);
    expect(wrapper).toMatchSnapshot();
});
