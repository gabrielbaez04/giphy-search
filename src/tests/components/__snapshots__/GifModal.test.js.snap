// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GifModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="gif-modal"
  closeTimeoutMS={0}
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <Slider
    gifs={
      Array [
        Object {
          "id": 1,
          "images": Object {
            "downsized_still": Object {
              "url": "test.url.com",
            },
          },
          "title": "test",
        },
        Object {
          "id": 2,
          "images": Object {
            "downsized_still": Object {
              "url": "test.url.com",
            },
          },
          "title": "test",
        },
        Object {
          "id": 3,
          "images": Object {
            "downsized_still": Object {
              "url": "test.url.com",
            },
          },
          "title": "test",
        },
      ]
    }
    selectedGif={0}
  />
</Modal>
`;
